Microsoft.Toolkit.Uwp.UI.Controls/WrapPanel/WrapPanel.cs

```diff
@@ -179,42 +198,69 @@ protected override Size MeasureOverride(Size availableSize)
         /// <inheritdoc />
         protected override Size ArrangeOverride(Size finalSize)
         {
-            var parentMeasure = new UvMeasure(Orientation, finalSize.Width, finalSize.Height);
-            var spacingMeasure = new UvMeasure(Orientation, HorizontalSpacing, VerticalSpacing);
-            var paddingStart = new UvMeasure(Orientation, Padding.Left, Padding.Top);
-            var paddingEnd = new UvMeasure(Orientation, Padding.Right, Padding.Bottom);
-            var position = new UvMeasure(Orientation, Padding.Left, Padding.Top);
-
-            double currentV = 0;
-            foreach (var child in Children)
+            if (Children.Count > 0)
             {
-                var desiredMeasure = new UvMeasure(Orientation, child.DesiredSize.Width, child.DesiredSize.Height);
-                if (desiredMeasure.U == 0)
-                {
-                    continue; // if an item is collapsed, avoid adding the spacing
-                }
+                var parentMeasure = new UvMeasure(Orientation, finalSize.Width, finalSize.Height);
+                var spacingMeasure = new UvMeasure(Orientation, HorizontalSpacing, VerticalSpacing);
+                var paddingStart = new UvMeasure(Orientation, Padding.Left, Padding.Top);
+                var paddingEnd = new UvMeasure(Orientation, Padding.Right, Padding.Bottom);
+                var position = new UvMeasure(Orientation, Padding.Left, Padding.Top);
 
-                if ((desiredMeasure.U + position.U + paddingEnd.U) > parentMeasure.U)
+                double currentV = 0;
+                void arrange(UIElement child, bool isLast = false)
```

Comment:
Do we want to pull this method out as a private one vs. inline?

We could use a ref parameter for the currentV?  Thoughts?

Reply:
Why? Are inline functions a no-no in the toolkit?

The only reason I see to change it is for project consistency.

Reply:
I don't think we have many, if any.  I just found it a bit harder to read as the looping through the children was below.  Not a big deal.

Reply:
Ok. I'm not changing it unless you force me to :P I really don't like the idea passing all of those props around just to un-nest the function.

Reply:
Ah, I forgot about all the other calculated properties, no worries then! 😊